marker_rustc_driver/src/conversion/marker/item.rs

```diff
@@ -115,6 +115,13 @@ impl<'ast, 'tcx> MarkerConverterInner<'ast, 'tcx> {
                             variant.disr_expr.map(|anon| self.to_const_expr(anon)),
                         )
                     }));
+                    // A block to limit the lifetime of the mut borrow
+                    {
+                        let mut cache = self.variants.borrow_mut();
+                        for var in variants {
+                            cache.insert(var.id(), var);
+                        }
```

Comment:
If you use `extend` instead of a for loop and the iterator returns the expected number of elements in the `size_hint()` method, then the map can pre-allocate enough space beforehand, meaning that `extend` is always faster than a for loop with `insert`, because it has more context, and won't do repeated allocations and even additional checks for growing the backing storage.
```suggestion
                        self.variants.borrow_mut().extend(variants.into_iter().map(|var| (var.id(), var)));
```

Same allies to all other for/insert loops added in the diff

Reply:
Yeah, you're absolutely right. There should be a Clippy lint for this ^^